### PR TITLE
[Shop][Account] Upgrade order history

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function changePaymentMethod(OrderInterface $order): void
             ['number' => $order->getNumber()],
         );
 
-        $link = $row->find('css', '[data-test-button="sylius.ui.pay"]');
+        $link = $row->find('css', '[data-test-button="sylius-pay"]');
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php]:
```suggestion
        $link = $row->find('css', '[data-test-button="pay"]');
```

File: src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ public function isOrderWithNumberInTheList($number): bool
 
     public function openLastOrderPage(): void
     {
-        $this->getElement('last_order')->find('css', '[data-test-button="sylius.ui.show"]')->click();
+        $this->getElement('last_order')->find('css', '[data-test-button="sylius-show"]')->click();
```
Comment (NoResponseMate) [inline src/Sylius/Behat/Page/Shop/Account/Order/IndexPage.php]:
```suggestion
        $this->getElement('last_order')->find('css', '[data-test-button="show"]')->click();
```

File: src/Sylius/Bundle/ShopBundle/templates/order/thank_you/payment_instruction.html.twig [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{% set lastPayment = hookable_metadata.context.order.payments.last() %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/order/thank_you/payment_instruction.html.twig]:
```suggestion
{% set last_payment = hookable_metadata.context.order.payments.last() %}
```
or just `payment`

File: src/Sylius/Bundle/ShopBundle/templates/account/order/grid/action/pay.html.twig [hunk 4]
```diff
@@ -3,5 +3,5 @@
 {% set path = options.link.url|default(path(options.link.route|default(grid.requestConfiguration.getRouteName('show')), options.link.parameters|default({'id': data.id}))) %}
 
 {% if data.paymentState in ['awaiting_payment'] %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/templates/account/order/grid/action/pay.html.twig]:
```suggestion
{% if data.paymentState == constant('Sylius\\Component\\Core\\OrderPaymentStates::STATE_AWAITING_PAYMENT') %}
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/services.xml [hunk 5]
```diff
@@ -38,7 +38,7 @@
 
         <service id="Sylius\Component\Core\Storage\CartStorageInterface" alias="sylius.storage.cart_session" />
 
-        <service id="sylius.grid_filter.shop_string" class="Sylius\Component\Grid\Filter\StringFilter">
+        <service id="sylius.grid_filter.shop_string" class="Sylius\Component\grid\filter\StringFilter">
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services.xml]: Should be reverted

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/shared/form/select_payment.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+            payment:
+                template: '@SyliusShop/shared/form/select_payment/payment.html.twig'
+        
+        'sylius_shop.shared.form.select_payment.payment':
+            header:
+                template: '@SyliusShop/shared/form/select_payment/payment/header.html.twig'
+            choice:
+                template: '@SyliusShop/shared/form/select_payment/payment/choice.html.twig'
+        
+        'sylius_shop.shared.form.select_payment.payment.#unavailable':
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/shared/form/select_payment.yaml]:
```suggestion
        'sylius_shop.shared.form.select_payment.payment#unavailable':
```

File: src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/shared/form/select_payment.yaml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+            unavailable:
+                template: '@SyliusShop/shared/form/select_payment/unavailable.html.twig'
+                
+        'sylius_shop.shared.form.select_payment.payment.choice':
+            details:
+                template: '@SyliusShop/shared/form/select_payment/payment/choice/details.html.twig'
+        
+        'sylius_shop.shared.form.select_payment.payment.choice.#unavailable':
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/shared/form/select_payment.yaml]:
```suggestion
        'sylius_shop.shared.form.select_payment.payment.choice#unavailable':
```

File: src/Sylius/Bundle/ShopBundle/templates/order/show/form.html.twig [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{% import '@SyliusShop/shared/messages.html.twig' as messages %}
+
+{% set order = hookable_metadata.context.order %}
+{% set form = hookable_metadata.context.form %}
+
+{% form_theme form '@SyliusShop/Form/theme.html.twig' %}
+
+{% if order.paymentState in ['awaiting_payment'] %}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/order/show/form.html.twig]: Similar to above suggestion, use constant here

File: src/Sylius/Bundle/ShopBundle/templates/order/show/form/pay_button.html.twig [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{% set order = hookable_metadata.context.order %}
+
+{% set order_cannot_be_paid = not sylius_order_can_be_paid(order) %}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/order/show/form/pay_button.html.twig]:
```suggestion
{% set order = hookable_metadata.context.order %}
{% set order_cannot_be_paid = not sylius_order_can_be_paid(order) %}

```

**Review by GSadee — APPROVED** [review]

**Pull request merged**